tenant-cd/src/main/java/ai/vespa/hosted/cd/http/HttpEndpoint.java

```diff
@@ -57,16 +59,24 @@ public URI uri() {
         return endpoint;
     }
 
+    @Override
+    public <T> HttpResponse<T> send(HttpRequest.Builder request, HttpResponse.BodyHandler<T> handler) {
+        try {
+            return client.send(authenticator.authenticated(request).build(), handler);
+        }
+        catch (IOException | InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     @Override
     public Search search(Query query) {
         try {
             URI target = endpoint.resolve(searchApiPath).resolve("?" + query.rawQuery());
-            HttpRequest request = authenticator.authenticated(HttpRequest.newBuilder()
-                                                                         .timeout(query.timeout().orElse(Duration.ofMillis(500))
-                                                                                       .plus(Duration.ofSeconds(1)))
-                                                                         .uri(target))
-                                               .build();
-            HttpResponse<byte[]> response = client.send(request, HttpResponse.BodyHandlers.ofByteArray());
+            HttpResponse<byte[]> response = send(HttpRequest.newBuilder(target)
+                                                            .timeout(query.timeout().orElse(Duration.ofMillis(500))
+                                                                          .plus(Duration.ofSeconds(1))),
```

Comment:
Is this standard practice Vespa? Seems backwards. If the client wants a timeout of X seconds they want the response no later than X seconds. To honor this the implementation/backends must use shorter timeouts to guarantee the response is sent no later than X.

Reply:
That would be good. Will experiment and see if it’s the case. Don’t know how the timeout relate to the events the server sees.